rust/osm2lanes/src/transform/tags_to_lanes/separator/semantic.rs

```diff
@@ -0,0 +1,79 @@
+use crate::metric::{Metre, Speed};
+use crate::road::{Designated, Style};
+use crate::transform::tags_to_lanes::Infer;
+
+/// Semantic speed class
+pub enum SpeedClass {
+    Walking,
+    /// ~30kph / ~20mph
+    Living,
+    /// ~50kph / ~30mph
+    Intra,
+    /// ~80kph / ~50mph
+    Inter,
+    Max,
+}
+
+impl From<Speed> for SpeedClass {
+    #[allow(clippy::panic)]
+    fn from(s: Speed) -> Self {
+        match s.kph() {
+            s if (0.0_f64..15.0_f64).contains(&s) => Self::Walking,
+            s if (15.0_f64..40.0_f64).contains(&s) => Self::Living,
+            s if (40.0_f64..70.0_f64).contains(&s) => Self::Intra,
+            s if (70.0_f64..100.0_f64).contains(&s) => Self::Inter,
+            s if (100.0_f64..200.0_f64).contains(&s) => Self::Max,
+            _ => panic!("unexpected speed {s}"),
+        }
+    }
+}
+
+/// Overtaking rules
+///
+/// Note: this does not take into account the local vehicle-specific rules,
+/// e.g. for motorcycle filtering or overtaking tractors.
+pub enum Overtaking {
+    Permitted,
+    _Prohibited,
+}
+
+impl Default for Overtaking {
+    fn default() -> Self {
+        // fail-deadly, see README
+        Self::Permitted
+    }
+}
+
+/// Semantic lane separator
+pub enum Separator {
+    /// Into grass or dirt
+    _SoftEdge,
+    /// Into a building or other hard surface
+    _HardEdge,
+    /// Motorway (or other) shoulder
+    Shoulder { speed: Infer<SpeedClass> },
+    /// Road paint between same direction
+    Lane {
+        speed: Infer<SpeedClass>,
+        overtaking: Overtaking,
+    },
+    /// Road paint between opposite direction
+    Centre {
+        speed: Infer<SpeedClass>,
+        overtaking: Overtaking,
+        more_than_2_lanes: bool,
+    },
+    /// Road paint between different modes
+    // TODO: solve directionality
+    Modal {
+        inside: Designated,
+        outside: Designated,
+    },
```

Comment:
I think only details that are properties of the separator should be part of this type, but the `separator to markings` transformations should have access to both adjacent lanes and the road. The speed can be different between lanes, but can that effect the marking style per lane? Also, I am sure that different locales use all sorts of exotic properties to determine marking styles. 

Then `Modal` could be removed in favour of `Lane` (possibly with `different_modes: bool` for convenience) and the directionality question could be dodged.

Reply:
>  only details that are properties of the separator should be part of this type

What is the semantic difference then between `Centre` vs `Lane` then?

to me there is a semantic difference between markings separating direction, markings separating lanes of the same type and direction, and markings separating modes?

Reply:
I see your point. A line separating a car lane from a bike lane *does* feel very different from a line separating two car lanes,  but a line separating a car lane from a bus lane is much more similar to the latter -- especially because bus lanes can often be used by cars to make turns or during part of the day. As long as `Centre` takes precedence,  `Modal` seems like a perfectly sensible distinction.

Reply:
> especially because bus lanes can often be used by cars to make turns or during part of the day

dabreegster will know that bus lanes in london are often shared between cyclists and busses and taxis.

Where I live all 3 modes are almost entirely always separated where possible.

I suggest sticking to our use of `designated` to mean the primary mode, and separators between those consisting only of paint should for a semantic category.